chapters/errors.Rmd

```diff
@@ -11,43 +11,37 @@ those programs might still contain bugs.
 We should therefore plan from the start to detect and handle errors.
 
 Something that goes wrong while a program is running
-is sometimes referred to as an \gref{exception}{exception} from normal behavior.\index{exception}
+is sometimes referred to as an exception from normal behavior.
 Generally speaking,
-we distinguish between \gref{internal errors}{internal_error},\index{error!internal}\index{internal error}
-such as calling a function with `None` instead of a list,
-and \gref{external errors}{external_error},
-such as trying to read a file that doesn't exist.
-Internal errors are mistakes in the program itself;
-they should be prevented by doing unit testing,
-but software is always used in new ways in the real world,
-and those new ways can trigger unanticipated bugs.
+we distinguish between two types of errors/exceptions.
+\gref{Internal errors}{internal_error},\index{error!internal}\index{internal error}
+are mistakes in the program itself,
+such as calling a function with `None` instead of a list.
+\gref{External errors}{external_error}\index{external error} are usually caused
+by interactions between the program and the outside world:
+a user may mis-type a filename,
+the network might be down,
+and so on.
 
 When an internal error occurs,
 the only thing we can do in most cases is report it and halt the program.
-If a function has been passed `None` instead of a valid list,
+For the function that has been passed `None` instead of a valid list,
```

Comment:
I dont' understand the use of "For the" instead of "If a" here?

Reply:
We only just mentioned that same example a few sentences earlier, so I was trying to avoid it sounding like we weren't aware of that... I agree that I haven't done a particularly good job, so I'll revert back to the original text.